compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/symbols/BInvokableTypeSymbol.java

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.ballerinalang.compiler.semantics.model.symbols;
+
+import org.ballerinalang.model.elements.PackageID;
+import org.ballerinalang.model.symbols.SymbolKind;
+import org.wso2.ballerinalang.compiler.semantics.model.types.BType;
+import org.wso2.ballerinalang.compiler.util.Names;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a function type symbol.
+ *
+ * @since 1.1.0
+ */
+public class BInvokableTypeSymbol extends BTypeSymbol {
```

Comment:
can't we use BInvokableSymbol? (or if this is sufficient remove that?)

Reply:
This was done thinking  separation of concerns. 

`BInvokableSymbol` is mapped to functions (with a body), while `BInvokableTypeSymbol` is mapped to function type descriptors. 

Every `BInvokableSymbol` has a type `BInvokableType` with a `BInvokableTypeSymbol`, but there can be `BInvokableTypeSymbol` that are not associated with a BInvokableSymbol.